src/components/DownloadInvoiceForm.js

```diff
@@ -40,16 +56,40 @@ export default class DownloadInvoiceForm extends Component {
     });
   }
 
-  handleClientChange(event) {
+  handleNamedStateChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleCheckboxChange = (event) => {
     this.setState({
-      client: event.target.value
+      showTaxes: event.target.checked
     });
   }
 
-  handleSubmit(event) {
+  handleRatesChange = (event, id) => {
+    const nextRate = parseInt(event.target.value, 10);
+    this.setState({
+     userRates: this.state.userRates.setIn([id, 'rate'], nextRate || 0)
+   });
+  }
+
+  handleSubmit = (event) => {
     event.preventDefault();
-    const { start, end, client } = this.state;
-    this.props.onSubmit(start, end, client);
+    const { start, end, client, address, notes, userRates, currency }
+      = this.state;
+    const userIdToRate = userRates.valueSeq()
```

Comment:
`rates-by-id` is a map of ids to rates. This is a list of objects. We're actually calling this userRates on the backend.